value-fixture/test/org/immutables/fixture/with/WithEnumsTest.java

```diff
@@ -0,0 +1,27 @@
+package org.immutables.fixture.with;
+
+import org.junit.Test;
+
+import java.math.RoundingMode;
+
+import static org.immutables.check.Checkers.check;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class WithEnumsTest {
+  private static final ImmutableWithEnums TEST_IMMUTABLE = ImmutableWithEnums.builder()
+          .roundingMode(RoundingMode.CEILING)
+          .maybeRoundingMode(RoundingMode.HALF_DOWN)
+          .nullableRoundingMode(RoundingMode.UNNECESSARY)
+          .build();
+
+  @Test
+  public void doNotAllowNullEnumValueInNonNullableWith() {
+    assertThrows(NullPointerException.class, () -> TEST_IMMUTABLE.withRoundingMode(null));
+  }
+
+  @Test
+  public void allowNullEnumValueInNullableWith() {
+    ImmutableWithEnums withNullableRoundingMode = TEST_IMMUTABLE.withNullableRoundingMode(null);
```

Comment:
Can you pls add tests which validate equals on enum values ?

For instance, populate `roundingMode` / `nullableRoundingMode` / `maybeRoundingMode` attributes with same / different / null values and ensure `WithEnums` instances are equal / different depending on enum values.

Reply:
Nice catch! I added a few tests to cover this in the latest revision.